eayunstack_tools/doctor/net.py

```diff
@@ -115,24 +123,41 @@ def port_log(device_owner, s):
                      % (device_owner, pid, l3_host))
 
         # 2) ping external gateway to check network status
+        #    and check external interface qos
         LOG.debug('check gateway for port on %s' % (l3_host))
         if device_owner == 'network:router_gateway':
             LOG.debug('this port is external port, check external gateway')
             gw = vrouter_get_gw_remote(l3_host, rid)
+            interface = vrouter_get_gw_interface(l3_host, rid)
             if gw:
                 LOG.debug("check external gateway %s on %s" % (gw, l3_host))
                 cmd = "ip netns exec qrouter-%s ping -c 1 %s" % (rid, gw)
                 out, err = ssh_connect(l3_host, cmd)
                 if not err:
                     LOG.debug("external gateway is ok")
                 else:
-                    LOG.error("failed to connect external gateway on %s" % (l3_host))
+                    LOG.error("failed to connect external gateway on %s"
+                              % (l3_host))
             else:
                 LOG.error("failed to get external gateway on %s" % (l3_host))
+            if interface:
+                LOG.debug("check external interface %s qos on %s"
+                          % (interface, l3_host))
+                cmd = ("ip netns exec qrouter-%s tc qdisc show dev %s"
+                       % (rid, interface))
+                out, err = ssh_connect(l3_host, cmd)
+                if out:
+                    LOG.debug("external interface %s qos is found on %s"
+                              % (interface, l3_host))
+                else:
+                    LOG.error("external interface %s qos is not found on %s"
```

Comment:
"qos was not found on external interface %s of qrouter-%s on %s" % (interface, rid, l3_host)